src/wp-includes/plugin.php

```diff
@@ -39,6 +39,38 @@
 if ( ! isset( $wp_current_filter ) )
 	$wp_current_filter = array();
 
+/**
+ * Check that the callback parameters match the number passed to add_action() or add_filter().
+ *
+ * @since 1.3.0
+ *
+ * @param $function_name   add_action() or add_filter(), as returned by __FUNCTION__.
+ * @param $function_to_add The callback to check.
+ * @param $accepted_args   The number of arguments claimed.
+ */
+function _check_callback_params($function_name, $function_to_add, $accepted_args)
+{
+    if ( defined( 'CP_STRICT' ) && CP_STRICT ) {
+        if ( is_string( $function_to_add ) && function_exists( $function_to_add ) ) {
+            $rf = new ReflectionFunction( $function_to_add );
+            $params = $rf->getParameters();
+            $numParams = count( $params );
```

Comment:
It seems like this can only ever be an array but it may still be an idea to do something like:
```suggestion
            $numParams =  ( is_array( $params ) || $params instanceof Countable ) ? count( $params ) : 0;
```

Reply:
It's a language construct, so I can't see how we'd get anything but an array back. If we're going to check it anyway then we should exit noisily as something has gone very wrong indeed :-)

Reply:
OK seems like it can't be anything other than an array and if it then other things are completely broken :)